### DISCUSSION TITLE
Parameter quoting in requests

### DISCUSSION THREAD

**Discussion by tobiasge** [encode/httpx#2320] [Q&A]
Hello,
I have the following request:
```python
params = {"filter[leaf]":{"op":"eq", "value":"true"}}
request = httpx.Request("GET", "/my/url", params=params)
```
When I look at the URL, I get this: `/my/url?filter%5Bleaf%5D=%7B%27op%27%3A+%27eq%27%2C+%27value%27%3A+%27true%27%7D`
Which decodes  to this: `/my/url?filter[leaf]={'op':+'eq',+'value':+'true'}`

Now the params in the URL are enclosed in single quotes and not in double quotes, which leads to an error in the API server:
`JSONDecodeError: Expecting property name enclosed in double quotes`

Any hints on how to get httpx to encode the params with double quotes?

**Comment by lovelydinosaur** [comment]
I'd suggest the following...

```python
filter = {"op":"eq", "value":"true"}
params = {"filter[leaf]": json.dumps(filter)}
```
